crates/runtime/src/instance/allocator/pooling/index_allocator.rs

```diff
@@ -0,0 +1,408 @@
+//! Index/slot allocator policies for the pooling allocator.
+
+use super::PoolingAllocationStrategy;
+use crate::CompiledModuleId;
+use rand::Rng;
+use std::collections::HashMap;
+
+#[derive(Clone, Debug)]
+pub(crate) enum PoolingAllocationState {
+    NextAvailable(Vec<usize>),
+    Random(Vec<usize>),
+    /// Reuse-affinity policy state.
+    ///
+    /// The data structures here deserve a little explanation:
+    ///
+    /// - free_list: this is a vec of slot indices that are free, no
+    ///   matter their affinities.
```

Comment:
This is *all* slot indices that are free, right? That is, an invariant of this code is that every free slot index is always in this list, right?

If my understanding is correct, then I think it makes sense to clarify this comment as such.

Reply:
Yep, clarified, thanks!